app/src/main/java/com/example/nasbackup/views/ConnectionSettingsViewModel.kt

```
@@ -0,0 +1,247 @@
+    }
+
+    fun testConnection() {
+        if (!_canTestConnection.value) return
+        _isTestingConnection.value = true
+        _isConnectionTestSuccessful.value = false
+
+        viewModelScope.launch(Dispatchers.IO) {
+            val success = try {
+                val smbFile = createSmbFile(null)
+                smbFile.exists()
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)